plugin-script-python/src/main/java/io/kestra/plugin/scripts/python/Commands.java

```diff
@@ -299,17 +294,26 @@ protected DockerOptions injectDefaults(RunContext runContext, DockerOptions orig
     public ScriptOutput run(RunContext runContext) throws Exception {
         TargetOS os = runContext.render(this.targetOS).as(TargetOS.class).orElse(null);
 
-        return this.commands(runContext)
+        boolean isCacheEnabled = runContext.render(this.dependencyCacheEnabled).as(Boolean.class).orElse(true);
+        ResolvedPythonEnvironment pythonEnvironment = setupPythonEnvironment(runContext, isCacheEnabled);
+
+        ScriptOutput output = this.commands(runContext)
             .addEnv(Map.of(
                 "PYTHONUNBUFFERED", "true",
                 "PIP_ROOT_USER_ACTION", "ignore",
-                "PIP_DISABLE_PIP_VERSION_CHECK", "1"
+                "PIP_DISABLE_PIP_VERSION_CHECK", "1",
+                "PYTHONPATH", pythonEnvironment.packages().path().toString()
```

Comment:
What if pip or uv is used as a before command?
Would it work like before if you set the PYTHONPATH?